backend/supabase/functions/test-local/index.ts

```
@@ -0,0 +1,309 @@
+  // Strip control characters and disallowed chars (allow alphanumeric, dash, underscore)
+  const studyGuideId = rawStudyGuideId.replace(/[^a-zA-Z0-9_-]/g, '')
+  if (studyGuideId.length === 0) {
+    return { error: 'study_guide_id contains only invalid characters' }
+  }
+
+  // Validate and sanitize question
+  if (rawQuestion.length > 500) {
+    return { error: 'question exceeds maximum length of 500 characters' }
+  }
+  // Strip control characters (keep alphanumeric, spaces, basic punctuation)
+  const question = rawQuestion.replace(/[\x00-\x1F\x7F]/g, '').trim()
```

Biome (2.1.2)

[error] Unexpected control character in a regular expression.

Control characters are unusual and potentially incorrect inputs, so they are disallowed.

(lint/suspicious/noControlCharactersInRegex)